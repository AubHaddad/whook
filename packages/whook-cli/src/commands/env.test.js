import initEnvCommand from './env';
import YError from 'yerror';

describe('envCommand', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should work', async () => {
    const envCommand = await initEnvCommand({
      log,
      ENV: { NODE_ENV: 'test' },
      args: {
        _: ['env'],
        name: 'NODE_ENV',
      },
    });
    const result = await envCommand();

    expect({
      result,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should work with a default value', async () => {
    const envCommand = await initEnvCommand({
      log,
      ENV: {},
      args: {
        _: ['env'],
        name: 'NODE_ENV',
        default: 'lol',
      },
    });
    const result = await envCommand();

    expect({
      result,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should fail with no value', async () => {
    const envCommand = await initEnvCommand({
      log,
      ENV: {},
      args: {
        _: ['env'],
        name: 'NODE_ENV',
      },
    });

    try {
      await envCommand();
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect({
        errorCode: err.code,
        errorParams: err.params,
        logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
      }).toMatchSnapshot();
    }
  });
});
