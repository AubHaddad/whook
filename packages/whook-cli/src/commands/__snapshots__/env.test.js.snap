// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`envCommand should fail with no value 1`] = `
Object {
  "errorCode": "E_NO_ENV_VALUE",
  "errorParams": Array [
    "NODE_ENV",
  ],
  "logCalls": Array [],
}
`;

exports[`envCommand should work 1`] = `
Object {
  "logCalls": Array [
    Array [
      "info",
      "test",
    ],
  ],
  "result": undefined,
}
`;

exports[`envCommand should work with a default value 1`] = `
Object {
  "logCalls": Array [
    Array [
      "info",
      "lol",
    ],
  ],
  "result": undefined,
}
`;
